### PR TITLE
Python: Promote `py/pam-auth-bypass`

### PR DESCRIPTION
**Pull request by RasmusWL**
This promotes the experimental query from https://github.com/github/codeql/pull/8595

### CODE REVIEW

File: python/ql/src/Security/CWE-285/PamAuthorization.ql [hunk 1]
```diff
@@ -1,36 +1,37 @@
...
~where
~  authenticateCall = libPam().getMember("pam_authenticate").getACall() and
~  handle = authenticateCall.getArg(0) and
~  not exists(API::CallNode acctMgmtCall |
     acctMgmtCall = libPam().getMember("pam_acct_mgmt").getACall() and
     DataFlow::localFlow(handle, acctMgmtCall.getArg(0))
   )
-select authenticateCall, "This PAM authentication call may be lead to an authorization bypass."
+select authenticateCall,
+  "This PAM authentication call may be lead to an authorization bypass, since 'pam_acct_mgmt' is not called afterwards."
```
[the remark this answers is not in the document]
Comment (RasmusWL) [reply python/ql/src/Security/CWE-285/PamAuthorization.ql]: fixed

File: python/ql/src/Security/CWE-285/PamAuthorization.ql [hunk 2]
```diff
@@ -33,4 +34,5 @@ where
     acctMgmtCall = libPam().getMember("pam_acct_mgmt").getACall() and
```
Comment (tausbn) [inline python/ql/src/Security/CWE-285/PamAuthorization.ql:34]: I'm not allowed to add a comment in the appropriate place, but there's a use of `API::moduleImport("ctypes.util")` on line 20 above that will need fixing.

**Review by tausbn — CHANGES_REQUESTED** [review]
A few minor bits and bobs that need fixing (most of them presented as suggestions for easy inclusion), but otherwise this looks good to me. :+1:

File: python/ql/src/Security/CWE-285/PamAuthorization.ql (hunk 2 above)
Comment (RasmusWL) [reply python/ql/src/Security/CWE-285/PamAuthorization.ql:34]: indeed, that probably what the merge conflict is also about :+1:

**Review by tausbn — APPROVED** [review]

**Pull request merged**